File: WaTF-Bank-iOS/watf-bank/PageController/AddFavourite.m
#import "Util.h"
#import "SqlCipher.h"

@interface AddFavourite : UIViewController <UITextFieldDelegate>
@property (weak, nonatomic) IBOutlet UITextField *accountNo;
@property (weak, nonatomic) IBOutlet UITextField *name;
@end

@implementation AddFavourite

- (void)viewDidLoad
{
    _accountNo.delegate = self;
}

- (BOOL)textField:(UITextField *)textField shouldChangeCharactersInRange:(NSRange)range replacementString:(NSString *)string
{
    NSCharacterSet *numbersOnly = [NSCharacterSet characterSetWithCharactersInString:@"0123456789"];
    NSCharacterSet *characterSetFromTextField = [NSCharacterSet characterSetWithCharactersInString:string];
    BOOL stringIsValid = [numbersOnly isSupersetOfSet:characterSetFromTextField];
    BOOL lengthIsValid = ([textField.text length]+[string length]>10);
    return stringIsValid && !lengthIsValid;
}

- (IBAction)push:(id)sender {
    if([SqlCipher insert:_name.text :_accountNo.text])
    {
        UIAlertController* alert = [UIAlertController
                                    alertControllerWithTitle:@"Add Favourite Success"
                                    message:nil
                                    preferredStyle:UIAlertControllerStyleAlert];
        UIAlertAction* ok = [UIAlertAction
                             actionWithTitle:@"OK"
                             style:UIAlertActionStyleDefault
                             handler:^(UIAlertAction * action)
                             {
                                 [Util changeView:self :@"Home"];
                             }];
        [alert addAction:ok];
        [self presentViewController:alert animated:YES completion:nil];
    }
    else
    {
        [Util alert:self :@"Add Favourite Failed"];
    }
}

@end

File: iOS-Swift/watf-bank/PageController/AddFavourite.swift
import UIKit

class AddFavourite : UIViewController, UITextFieldDelegate {
    
    @IBOutlet weak var name: UITextField!
    @IBOutlet weak var accountNo: UITextField!
    
    override func viewDidLoad() {
        accountNo.delegate = self
    }
    
    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        name.resignFirstResponder()
        accountNo.resignFirstResponder()
    }
    
    func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
        let allowChar = CharacterSet(charactersIn: "0123456789")
        return  allowChar.isSuperset(of: CharacterSet(charactersIn: string)) && (textField.text!.count+string.count)<=10
    }
    
    @IBAction func push(_ sender: Any) {
        if SQLCipher.insertFav(name.text!, accountNo.text!) {
            let a = UIAlertController(title: "Add Succeeded", message: nil, preferredStyle: .alert)
            a.addAction(UIAlertAction(title: "OK", style: .default, handler: {(ac) in
                    Util.changeView(self, "Home")
                }))
            self.present(a, animated: true, completion: nil)
        }
        else {
            Util.alert(self, "Add Failed")
        }
    }
}
